### DISCUSSION TITLE
Attach metadata to PR (other than status), or close / comment without notification?

### DISCUSSION THREAD

**Discussion by xmo-odoo** [community/community#24828] [Apps, API and Webhooks]
<p>When using an external tool to handle the PR integration process (see: <a href="https://graydon2.dreamwidth.org/1597.html" rel="nofollow noopener">Not Rocket Science Rule to Software Engineering</a>) the external tool has to close the PR explicitly and <em>should</em> probably indicate where / how the PR was integrated (e.g. the hash of the merge / rebase / squash commit).</p>
<p>Via the API, this requires two calls:</p>
<pre><code>PATCH /repos/:owner/:repo/pulls/:pull_number
</code></pre>
<p> to change its state to closed then</p>
<pre><code>POST /repos/:owner/:repo/issues/:issue_number/comment
</code></pre>
<p>to add a comment providing relevant information.</p>
<p>This, in turn, generates two notifications to every follower of the PR for what is essentially a single event. On busy repositories, this generates a lot of spam (even more so when using a tool like CPython’s <a href="https://github.com/python/miss-islington" rel="nofollow noopener">Miss Islington</a> which semi-automatically creates backports for PRs, backports onto which the original contributors likely get automatically pinged. For instance <a href="https://github.com/python/cpython/pull/16597" rel="nofollow noopener">python/cpython#16597</a> tags the original contributor &amp; the original reviewer in comments, leading to them getting all messages by default, and thus a lot of churn. Even more so as PR notifications / pings tend to be relevant / high-SNR information (people or colleagues bringing up issues or asking for information / feedback) and would thus be amongst the last to be binned or web-only.</p>
<p>Alternatively, I’m also open to an other way (than a comment) of adding relevant information to the PR, edirting an existing comment (or the PR message) feels distasteful and adding such meta-information as status seems like a misuse of the feature, not to mention statuses apparently disappear / get suppressed when a PR is closed or merged which rather defeats the intent as we specifically want to tell users where the pull request was actually integrated into the history proper.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>I’m not entirely certain what you mean by “integrated” since that is a fairly overloaded term. But you may want to look at the <a href="https://developer.github.com/v3/repos/deployments/" rel="nofollow noopener">Deployments API</a> if you haven’t already? It sounds like what you want is some sort of “after the PR is merged” process, so this might be what you’re looking for?</p>
<p>Let us know if you have more questions.</p>

**Comment by xmo-odoo** [comment]
<p>That does look extremely interesting, are deployments visible but silent to followers of the PR (aka some sort of link is visible on the PR, but the information is not pushed to followers)?</p>
<p>If that’s the case, not only would that suit my needs perfectly, it might provide QoL improvements to various ancillary tasks. I can’t believe I missed this thing.</p>

**Comment by xmo-odoo** [comment]
<p><span class="mention">@lee-dohmafter</span> some testing deployments do seem to fit the bill, however some bits are a bit unclear e.g. the descriptions (on deployment &amp; deployment statuses) don’t seem to be used anywhere, and only the “environment”, “state” and “target_url” look relevant from an UI perspective (rather than automated tooling), is that correct?</p>

**Comment by lee-dohm** [comment]
<p>To be honest, I don’t know if all the fields are used in the UI or not. I believe the description would show up in the <a href="https://github.com/lee-dohm/atom-style-tweaks/deployments" rel="nofollow noopener">deployments UI </a>for the repository, if used. None of the tools that I’ve employed use the description field though, so I’m not certain.</p>
